scripts/format-ros2.sh

```
@@ -110,110 +116,251 @@ parse_args() {
+            return 0
+        fi
+    fi
+}
 
-        # Check if file would be changed by clang-format
-        if ! clang-format --dry-run --Werror "$file" >/dev/null 2>&1; then
+# Get expected header guard for a file
+# Pattern: PACKAGE__FILENAME_HPP_
+get_expected_guard() {
+    local file="$1"
+    local rel_path="${file#$ROS2_DIR/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)